tests/endtoend/test_http_functions.py

```diff
@@ -220,18 +224,160 @@ def tearDownClass(cls):
+
+    @testutils.retryable_test(3, 5)
+    def test_return_html(self):
+        """Test if the return_html function returns an HTML response"""
+        root_url = self.webhost._addr
+        html_url = f'{root_url}/api/return_html'
+        r = requests.get(html_url, timeout=REQUEST_TIMEOUT_SEC)
+        self.assertTrue(r.ok)
+        self.assertEqual(r.headers['content-type'],
+                         'text/html; charset=utf-8')
+        # Validate HTML content
+        expected_html = "<html><body><h1>Hello, World!</h1></body></html>"
+        self.assertEqual(r.text, expected_html)
+
+    @testutils.retryable_test(3, 5)
+    def test_return_ujson(self):
+        """Test if the return_ujson function returns a UJSON response"""
+        root_url = self.webhost._addr
+        ujson_url = f'{root_url}/api/return_ujson'
+        r = requests.get(ujson_url, timeout=REQUEST_TIMEOUT_SEC)
+        self.assertTrue(r.ok)
+        self.assertEqual(r.headers['content-type'], 'application/json')
+        self.assertEqual(r.text, '{"message":"Hello, World!"}')
+
+    @testutils.retryable_test(3, 5)
+    def test_return_orjson(self):
+        """Test if the return_orjson function returns an ORJSON response"""
+        root_url = self.webhost._addr
+        orjson_url = f'{root_url}/api/return_orjson'
+        r = requests.get(orjson_url, timeout=REQUEST_TIMEOUT_SEC)
+        self.assertTrue(r.ok)
+        self.assertEqual(r.headers['content-type'], 'application/json')
+        self.assertEqual(r.text, '{"message":"Hello, World!"}')
+
+    @testutils.retryable_test(3, 5)
+    def test_return_file(self):
+        """Test if the return_file function returns a file response"""
+        root_url = self.webhost._addr
+        file_url = f'{root_url}/api/return_file'
+        r = requests.get(file_url, timeout=REQUEST_TIMEOUT_SEC)
+        self.assertTrue(r.ok)
+        self.assertIn('@app.route(route="default_template")', r.text)
+
+    @testutils.retryable_test(3, 5)
+    def test_upload_data_stream(self):
+        """Test if the upload_data_stream function receives streaming data
+        and returns the complete data"""
+        root_url = self.webhost._addr
+        upload_url = f'{root_url}/api/upload_data_stream'
+
+        # Define the streaming data
+        data_chunks = [b"First chunk\n", b"Second chunk\n"]
+
+        # Define a function to simulate streaming by reading from an
+        # iterator
+        def stream_data(data_chunks):
+            for chunk in data_chunks:
+                yield chunk
+
+        # Send a POST request with streaming data
+        r = requests.post(upload_url, data=stream_data(data_chunks))
+
+        # Assert that the request was successful
+        self.assertTrue(r.ok)
+
+        # Assert that the response content matches the concatenation of
+        # all data chunks
+        complete_data = b"".join(data_chunks)
+        self.assertEqual(r.content, complete_data)
+
+    @testutils.retryable_test(3, 5)
```

Comment:
Remove the retries if its needed

Reply:
i think its good to have them as we still have flaky test cases. will update later once we have xdist fixed